### DISCUSSION TITLE
User inactivity

### DISCUSSION THREAD

**Discussion by joaobibiano** [vercel/swr#862] [Q&A] [answered]
Hi,

I searched on the documentation, but I didn't find anything about stop fetching when detect user inactivity (user stops to move their mouse during some time) 

Do you knows something about that?

Thank you

**Comment by huozhi** [comment]
@joaovbibiano we're working on feature now to let user stop revalidations under specific condition. and the user inactivity detection will be implemented on user side since the condition is fully customized.

will uodate the doc later once the new feature is released.

**Reply by joaobibiano** [reply]
@huozhi  many thanks! you guys are awesome!

**Comment by joaobibiano** [comment, marked as answer]
For those who need implent this, I would like to suggest a temporary solution like this

```
 const userIsInactive = useIdle(TIME_TO_CONSIDER_USER_INACTIVE);

 const { data: tabCounters } = useSWR(
  userIsInactive ? null : "request-key",
  fetcher,
  {
    refreshInterval: TIME_TO_FETCH_COUNTERS
  }
);
```

You can find the hook `useIdle` here [https://github.com/streamich/react-use/blob/master/docs/useIdle.md](React-use)

**Comment by huozhi** [comment]
swr@0.4.x is released with a new config `config.isPaused(): boolean`, you can let it return true when it's in idle state.

```js
const isIdle = useIdle(3e3);
useSWR(api, fetcher, {isPaused: () => isIdle})
```

**Reply by joaobibiano** [reply]
@huozhi why did you use "useCallback" here? Because isIdle keeps the same until it changes it's state, rigth?

**Reply by huozhi** [reply]
@joaovbibiano ya thanks, update it to make it shorter. `isPaused` suppose to work well with swr@4.0.1
